tests/collections/nlp/test_chat_sft_dataset.py

```diff
@@ -0,0 +1,317 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import json
+import os
+import random
+
+import pytest
+
+from nemo.collections.nlp.data.language_modeling.megatron.gpt_sft_chat_dataset import GPTSFTChatDataset
+from nemo.collections.nlp.modules.common.tokenizer_utils import get_nmt_tokenizer
+
+TOKENIZER_FILE_43B = '/home/TestData/nlp/megatron_sft/tokenizer.model'
+MERGE_FILE = '/home/TestData/nlp/megatron_sft/merges.txt'
+VOCAB_FILE = '/home/TestData/nlp/megatron_sft/vocab.json'
+
+
+def ids_to_text(tokenizer, ids):
+    tokens = tokenizer.ids_to_tokens(ids)
+    text = tokenizer.tokens_to_text(tokens)
+    return text
+
+
+def get_random_sentence():
+    nouns = ("puppy", "car", "rabbit", "girl", "monkey")
+    verbs = ("runs", "hits", "jumps", "drives", "barfs")
+    adv = ("crazily.", "dutifully.", "foolishly.", "merrily.", "occasionally.")
+    num1 = random.randrange(0, 5)
+    num2 = random.randrange(0, 5)
+    num3 = random.randrange(0, 5)
+    return nouns[num1] + ' ' + verbs[num2] + ' ' + adv[num3]
+
+
+def get_random_label():
+    keys = ["quality", "toxicity", "humor", "creativity", "violence", "helpfulness", "not_appropriate"]
+    values = [random.randrange(0, 5) for i in range(len(keys))]
+    return ",".join([k + ":" + str(v) for k, v in zip(keys, values)])
+
+
+def create_data_points(mask_user, turn_num, records, temp_file, t2v, label=True):
+    data_points = []
+    with open(temp_file, 'w', encoding='utf-8') as f:
+        for r in range(records):
+            record = {}
+            record['system'] = 'a chat\n\n'
+            record['type'] = 'TEXT_TO_VALUE' if t2v else 'VALUE_TO_TEXT'
+            record['mask'] = 'User' if mask_user else 'Assistant'
+            turns = []
+            record['conversations'] = turns
+            for i in range(turn_num):
+                turn = {}
+                turn['from'] = 'User' if i % 2 == 0 else 'Assistant'
+                turn['value'] = get_random_sentence()
+                if label:
+                    turn['label'] = get_random_label()
+                turns.append(turn)
+            f.write(json.dumps(record, ensure_ascii=False) + '\n')
+            data_points.append(record)
+    return data_points
+
+
+class TestGPTSFTChatDataset:
+    @classmethod
+    def setup_class(cls):
+        pass
+
+    @pytest.mark.unit
+    def test_43B_tokenizer_mask_user(self):
+        random.seed(5)
+        temp_file = '/tmp/test_file.jsonl'
+        turn_num = 5
+        records = 5
+        try:
```

Comment:
why are we doing this inside a try block?

Reply:
so we can make sure to remove the file in the `finally`